### ISSUE TITLE
Convert project to be usable with GNU Compiler Collection (GCC)?

### ISSUE THREAD

**Issue by piorrro33** [piorrro33/psxjin#61]
For our project to work with CircleCI (which makes tests and builds our code), we'll have to do some modifications to get our code to work with GCC.
Since I don't know how to do that AT ALL, I thought it would be a good thing to make an issue and se what could be done. And if you are ok with this.

**Comment by vgturtle127** [comment]
I can take a look after we get it to compile with make. Although isn't make part of the GCC? Maybe I'm insane or something but I thought that was how it worked. :stuck_out_tongue_winking_eye:

**Comment by piorrro33** [comment]
Yeah, gcc = make because in the Makefile, you put gcc commands :smiley:

**Comment by vgturtle127** [comment]
Alright, I thought that was it. :stuck_out_tongue_closed_eyes: 

I am OK with things like TravisCI and CircleCI, but I have never used anything like that before on my own projects. So I can't really help much in that regard.

But in regards to the makefile, I can probably get it to work once Lua quits being stupid. :-1: 

Also, I'm not sure you are aware, but there is a lot of Win32-specific files and Win32 code used in the project. So it most likely not compile on Linux or Mac for awhile at least. Just FYI. :open_mouth:

**Comment by piorrro33** [comment]
In fact, just to get CircleCI / TravisCI to work, we just need to make it 
buildable on Linux. I'll see if it's possible to do anything about these 
DLL's, to turn them into OBJ's.

**Comment by vgturtle127** [comment]
Oh. I thought they would work even if it just built for Windows?

Is TravisCI/CircleCI just for Linux?

**Comment by piorrro33** [comment]
Well, it is not just for Linux, but the machine that builds the code is running Linux.

**Comment by piorrro33** [comment]
By the way, you did an insane job on the last commit. Thank you for everything!

**Comment by vgturtle127** [comment]
No problem, man. :smiley: 

Good to know about the building on Linux, that makes more sense.
